tests/test_curves.py

```diff
@@ -820,3 +820,35 @@ def test_compute_ndarray_error(self):
         c = SplineXYZCurve(X=R * np.cos(phi), Y=R * np.sin(phi), Z=np.zeros_like(phi))
         with pytest.raises(TypeError):
             c.compute("length", grid=np.linspace(0, 1, 10))
+
+    @pytest.mark.unit
+    def test_discontinuous_knots(self):
+        """Test with discontinuous knots."""
+        discontinuous_indices = [0, 500]
+        R = 2
+        phi = 2 * np.pi * np.linspace(0, 1, 1001, endpoint=True) ** 2
+        discontinuous = SplineXYZCurve(
+            X=R * np.cos(phi),
+            Y=R * np.sin(phi),
+            Z=np.zeros_like(phi),
+            knots="arclength",
+            discontinuous_indices=discontinuous_indices,
+            method="linear",
+        )
+
+        continuous = SplineXYZCurve(
+            X=R * np.cos(phi),
+            Y=R * np.sin(phi),
+            Z=np.zeros_like(phi),
+            knots="arclength",
+            discontinuous_indices=None,
+            method="linear",
+        )
+
+        assert discontinuous.method == "linear"
+        assert continuous.method == "linear"
+
+        discont_torsion = discontinuous.compute("torsion")["torsion"]
+        cont_torsion = continuous.compute("torsion")["torsion"]
+
+        np.testing.assert_allclose(discont_torsion, cont_torsion, rtol=1e-3, atol=1e-3)
```

Comment:
Since the method is linear, isnt the torsion zero, since the second derivative is 0 and torsion involves the curve's second derivative?

A better test would be length (which should be pretty close if not exactly identical if you are using linear interpolation). I am not sure of a good test with a derivative quantity but you could try using curvature I guess (which for linear might be the same again)

And a test where the inteprolation method is not linear might be good too to actually test the derivatives computation, though I dont have a good one off the top of my head